core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -350,6 +356,10 @@ class BlockManagerMasterEndpoint(
     ).map(_.flatten.toSeq)
   }
 
+  private def externalShuffleServiceIdOnHost(blockManagerId: BlockManagerId): BlockManagerId = {
+    BlockManagerId(blockManagerId.executorId, blockManagerId.host, externalShuffleServicePort)
```

Comment:
I had to step through this pretty carefully to figure out why you were passing in `blockManagerId.executorId` here, though it was actually owned by the shuffle service.  In fact I was originally thinking that you should put in a "dummy" executor Id in here so it was clear its actually referring to something owned by the shuffle service.  But now I see that you actually need to know the original executor, to know which dirs to look in.  It might be worth at least a comment here that we need to keep a handle on the original executor, so the shuffle service knows which dir to look in.